src/core/src/preprocess/color_utils.cpp

```diff
@@ -20,8 +20,10 @@ std::unique_ptr<ColorFormatInfo> ColorFormatInfo::get(ColorFormat format) {
         res.reset(new ColorFormatInfoI420_ThreePlanes(format));
         break;
     case ColorFormat::RGB:
+        res.reset(new ColorFormatRGB(format));
+        break;
     case ColorFormat::BGR:
-        res.reset(new ColorFormatNHWC(format));
+        res.reset(new ColorFormatBGR(format));
```

Comment:
Why do you change something in RGB/BGR logic?

Reply:
`ColorFormatBGR` contains specific computation of tensor shape - https://github.com/openvinotoolkit/openvino/pull/13855/files#diff-49bd0f7677452175eb6f598dfb1a537321692e488278c206b8a15d363993ebbcR104